rag/utils/text_processing.py

```diff
@@ -0,0 +1,44 @@
+from rag.utils import Document
+
+from langchain_text_splitters.nltk import NLTKTextSplitter
+
+from typing import List
+
+
+def nltk_splitter(
```

Comment:
Another method to try out here would be some semantic chunking and see if it improves the performance.